### ISSUE TITLE
Parameter 'X' must be one of AllowedValues

### ISSUE THREAD

**Issue by stleon** [aws-amplify/amplify-cli#6224]
**Describe the bug**

Seems like paremeters from paremeters.json are not passed to the cloudformation template.

**Amplify CLI Version**
4.40.0

**To Reproduce**
I want to use [Custom CloudFormation stacks](https://docs.amplify.aws/cli/usage/customcf).

parameters.json:

```
{
    "X": "Yes"
}
```

category-cloudformation-template.yaml:

```
Parameters:
  env:
    Description:  the env parameter will be passed in to your cloudformation templates dynamically by the CLI
    Type: String

  X:
    Description: some description
    Type: String
    Default: No
    AllowedValues:
      - Yes
      - No
...
```

Then I run:

```
$ amplify env checkout dev
$ amplify push -y
```

And I have error:

```
CREATE_FAILED      xxxx          AWS::CloudFormation::Stack Sun Dec 20 2020 15:24:00 GMT+0300  Parameter 'X' must be one of AllowedValues
```

**Expected behavior**
Successful adding of custom CloudFormation stack.

**Desktop (please complete the following information):**
 - OS: Mac
 - Node Version: v15.3.0

**Comment by akshbhu** [comment]
Hi @stleon 

I guess there is a problem in category-cloudformation-template.yaml file when you are referencing the "X" parameter.
you can use ```Ref``` to use the value for "X" in parameters.json.

**Comment by stleon** [comment]
Hi @akshbhu 

I do not need use `Ref`. I want to receive such parameters as `ENABLED/DISABLED/PREFERRED` (strings) in template.

For example, In s3-cloudformation-template.json:
```
"triggerFunction": {
			"Type": "String"
		}
```
in parameters.json:
```
"triggerFunction": "NONE"
```

Not `Ref`.

**Comment by rajrajhans** [comment]
Hi @stleon. I tried to reproduce your issue, but I was able to pass the parameter to CloudFormation, and it showed up in my AWS Console as well. 

I believe the issue is with your YAML template. In the `category-cloudformation-template.yaml` which you have posted, it should be `'Yes'` and `'No'` instead of just `Yes` and `No`. Check the screenshots below for reference.

This works:
![image](https://user-images.githubusercontent.com/32734049/107869518-1aa3a400-6eb5-11eb-8ed6-1ff7dc29af5e.png)


This doesn't:
![image](https://user-images.githubusercontent.com/32734049/107869508-ff389900-6eb4-11eb-8ba0-6509c6b95da6.png)
